docs/roman/datamodels/models.rst

```diff
@@ -8,58 +8,84 @@ the underlying file format.  The same data model may be used for data
+arrays associated with it:
 
     - `data`: The science data
     - `dq`: The data quality array
     - `err`: The error array
 
-The shape of these arrays must be broadcast-compatible.  If you try to
-assign an array to one of these members that is not
-broadcast-compatible with the data array, an exception is raised.
+The shape of these arrays must be broadcast-compatible. If you try to
+assign an array to one of these members that is not broadcast-compatible with
+the data array, an exception is raised.
+
+Along with data arrays the datamodel also contains information about the
+observation that can include the Observation program, exposure information,
+pointing informtation and processing steps.
 
 Working with models
 ===================
 
+Reading a data model
+--------------------
+
+If you have an exisiting data file it is straightforward to access the file
+using python.
+
+.. code-block:: python
+
+    from roman_datamodels import datamodels as rdm
+    fn = 'r0019106003005004023_03203_0034_WFI01_cal.asdf'
+    data_file = rdm.open(fn)
+    type(data_file)
+
+Where the output of the type command tells you that you have imported an
+ImageModel from roman_datamodels,
+
+
+    <class 'roman_datamodels.datamodels.ImageModel'>
+
 Creating a data model from scratch
 ----------------------------------
 
 To create a new `ImageModel`, just call its constructor.  To create a
 new model where all of the arrays will have default values, simply
 provide a shape as the first argument::
 
-.. doctest-skip::
+    from roman_datamodels import datamodels as rdm
+    from roman_datamodels.testing.factories import create_wfi_image
+    import numpy as np
 
-    from romancal.datamodels import ImageModel
-    with ImageModel((1024, 1024)) as im:
+    image_node = create_wfi_image()
+    image_model = rdm.ImageModel(image_node)
+    type(image_model)
+    #<class 'roman_datamodels.datamodels.ImageModel'>
+
+.. note ::
+    The values in the file generated by create_wfi_image are intended to be
+    clearly incorrect and should be replaced if the file is intended to be used
+    for anything besides a demonstration.
 
-        ...
 
 In this form, the memory for the arrays will not be allocated until
 the arrays are accessed.  This is useful if, for example, you don’t
 need a data quality array -- the memory for such an array will not be
 consumed::
 
   # Print out the data array.  It is allocated here on first access
-  # and defaults to being filled with zeros.
-  print(im.data)
+  # and defaults to being filled with random numbers.
+  print(image_model.data)
 
 If you already have data in a numpy array, you can also create a model
 using that array by passing it in as a data keyword argument::
 
-    data = np.empty((50, 50))
-    dq = np.empty((50, 50))
+    data = np.empty((50, 50),dtype=np.float32)
+    dq = np.empty((50, 50),dtype=np.uint32)
     with ImageModel(data=data, dq=dq) as im:
```

Comment:
Does this work for roman_datamodels?  I thought the ImageModel had to be initialized with a WfiImage object.

Reply:
Well whatever I did to make this work before is beyond me. I'm taking this out until I can work out the details.